bundles/org.eclipse.search/search/org/eclipse/search/internal/ui/SearchPlugin.java

```diff
@@ -91,15 +93,14 @@ public class SearchPlugin extends AbstractUIPlugin {
 
 	private List<SearchPageDescriptor> fPageDescriptors;
 	private List<SorterDescriptor> fSorterDescriptors;
-	private TextSearchEngineRegistry fTextSearchEngineRegistry;
 	private TextSearchQueryProviderRegistry fTextSearchQueryProviderRegistry;
 
 	public SearchPlugin() {
 		super();
 		Assert.isTrue(fgSearchPlugin == null);
 		fgSearchPlugin= this;
-		fTextSearchEngineRegistry= null;
 		fTextSearchQueryProviderRegistry= null;
+		SearchCorePlugin.getDefault().setDirtyFileDiscovery(new DirtyFileSearchParticipant());
```

Comment:
So this will only happen once the UI bundle is started. Is this the right direction for the dependency, or should this contribution happen even without search UI started? My first impression is that we want the later: the dirtyFileSearchParticipant can participate to search even in case the Search UI bundle wasn't started (because no search UI class was loaded yet).
If it is so, the best approach would be to make the DirtyFileSearchParticipant and OSGi service, and request for an optional service implementing IDirtyFileSearchParticipant in SearchCorePlugin.

Reply:
Please note that requesting that service will then activate the SearchPlugin (this must not be bad per-se), but I think if its made a service one should better think about if not even *multiple* such services would be possible to support!

Reply:
I've updated the code to theoretically support multiple services, but current implementation only chooses that with the highest weight. 

I'm really not in the mood to try to merge different versions of dirty files if, for example, two services return the same IFile with different contents. And I certainly don't want to change or break the actual search logic. So to avoid all this, a service provides a weight, and only the highest weight is chosen as the service in charge.